src/core/worker.js

```diff
@@ -671,11 +671,9 @@ var WorkerMessageHandler = {
 
       var onFailure = function(e) {
         if (e instanceof PasswordException) {
-          if (e.code === PasswordResponses.NEED_PASSWORD) {
-            handler.send('NeedPassword', e);
-          } else if (e.code === PasswordResponses.INCORRECT_PASSWORD) {
-            handler.send('IncorrectPassword', e);
-          }
+          handler.sendWithPromise('Password', e).then(function (data) {
+            pdfManager.updatePassword(data.password);
+          });
```

Comment:
add `.catch(function (e) { handler.send('PasswordException', e); });`. (I'm not sure if we need to re-wrap `e`, but I think this will work out) We need to add 'PasswordException' handler, to abort loading task. `Password` can be `PasswordRequest` to avoid confusion.